pkg/grpc/api.go

```diff
@@ -365,6 +366,112 @@ func (a *api) CallActor(ctx context.Context, in *internalv1pb.InternalInvokeRequ
 	return resp.Proto(), nil
 }
 
+func (a *api) PublishActorEvent(ctx context.Context, in *runtimev1pb.PublishActorEventRequest) (*emptypb.Empty, error) {
+	if a.pubsubAdapter == nil {
```

Comment:
@halspang can you think of a way to avoid code duplication here?

```
publishEventInternal(ctx context.Context, in interface{})
{

    switch in.(type) {
        case *runtimev1pb.PublishActorEventRequest: {
          // Actor publish path
        }
        case *runtimev1pb.PublishEventRequest: {
          // Regular publish path
        }
    }
}
```

or some generic way?

Essentially we should have the same code somehow for everything other than the addition of cloudevent fields.

Reply:
Given that everything is the same except for the two new fields, I'd think you could just use a single common method. In that method either choose to add the fields if they're defined or, just add them anyway as they'll be null in the message. 

That last approach does add two attributes to all requests though, which I'm not sure everyone would be happy about so I'd go with the first approach.